firefly-api/src/models.rs

```diff
@@ -19,3 +22,35 @@ pub mod casper {
         tonic::include_proto!("casper.v1");
     }
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TransferResult {
+    pub cost: u64,
+    pub errored: bool,
+    pub system_deploy_error: Option<String>,
+}
+
+impl TransferResult {
+    pub fn new(block_data: Value) -> anyhow::Result<Self> {
+        let cost = block_data
+            .get("cost")
+            .and_then(Value::as_u64)
+            .expect("cost not found");
```

Comment:
I think you meant `context` instead of `expect`.
```suggestion
            .context("cost not found")?;
```

Reply:
ok